code/modules/mob/living/basic/trooper/trooper_ai.dm

```diff
@@ -0,0 +1,129 @@
+/datum/ai_controller/basic_controller/trooper
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic/attack_until_dead
+	)
+
+	ai_movement = /datum/ai_movement/basic_avoidance
+	idle_behavior = /datum/idle_behavior/idle_random_walk
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/attack_obstacle_in_path/trooper,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree,
+		/datum/ai_planning_subtree/travel_to_point/and_clear_target/reinforce,
+	)
+
+/datum/ai_planning_subtree/basic_melee_attack_subtree/trooper
+	melee_attack_behavior = /datum/ai_behavior/basic_melee_attack/trooper
+
+/datum/ai_behavior/basic_melee_attack/trooper
+	action_cooldown = 1.2 SECONDS
+
+/datum/ai_planning_subtree/attack_obstacle_in_path/trooper
+	attack_behaviour = /datum/ai_behavior/attack_obstructions/trooper
+
+/datum/ai_behavior/attack_obstructions/trooper
+	action_cooldown = 1.2 SECONDS
+
+/datum/ai_controller/basic_controller/trooper/calls_reinforcements
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/call_reinforcements/nanotrasen,
+		/datum/ai_planning_subtree/attack_obstacle_in_path/trooper,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree,
+		/datum/ai_planning_subtree/travel_to_point/and_clear_target/reinforce,
+	)
+
+/datum/ai_controller/basic_controller/trooper/peaceful
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/target_retaliate,
+		/datum/ai_planning_subtree/call_reinforcements/nanotrasen,
+		/datum/ai_planning_subtree/attack_obstacle_in_path/trooper,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree,
+		/datum/ai_planning_subtree/travel_to_point/and_clear_target/reinforce,
+	)
+
+/datum/ai_controller/basic_controller/trooper/ranged
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/maintain_distance,
+		/datum/ai_planning_subtree/basic_ranged_attack_subtree/trooper,
```

Comment:
You are getting occasional runtimes because you are trying to simultaneously run two subtrees which set movement targets, which annoyingly often results in the error you are experiencing. Can't quite remember why. 
If some other behaviour is controlling what distance you are at from the target, use `/datum/ai_planning_subtree/ranged_skirmish` instead.

Reply:
Alternately remove this stack trace and just silently don't run the behaviour, relying on devs to manage it properly. It's sort of a false positive in this case.

Reply:
I've decided to remove the maintain distance behavior, because it wasn't working very effectively in spite of the runtimes and the basic Syndicate mobs didn't use it to begin with. I think it'd be _cool_ if the ranged troopers were a little smarter about keeping away, but I feel like that could be a balance change for later.

(Also, converting to ranged skirmish would be a hefty amount of refactoring work, which made me not want to do it right now.)

If it's preferable to figure it out instead, I can still do that, but I don't think it's strictly necessary to this conversion.